### ISSUE TITLE
RPTRST Not Really Recognised in SOLUTION Section

### ISSUE THREAD

**Issue by bska** [OPM/opm-common#1862]
I noticed that our test case [MOD4_GRP](https://github.com/OPM/opm-tests/blob/c95c363e87437e22f9ad12b72ba48f3d5d955169/model4/MOD4_GRP.DATA) appears not to write restart files for report steps other than the initial condition.  It **appears** that [`RestartConfig::getWriteRestartFile()`](https://github.com/OPM/opm-common/blob/643ff6e777b894f9bf7e641b92b9b1fbfb8bec03/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.hpp#L340) misses important aspects of `RPTRST` when that keyword is placed in the SOLUTION section.

I can reproduce the behaviour with the synthetic unit test below:
```
${ROOT}/opm-common/tests/parser/RestartConfigTests.cpp(1135): error: in "RPTRST_AND_RPTSOL_SOLUTION": Must write restart information for included step 1
${ROOT}/opm-common/tests/parser/RestartConfigTests.cpp(1135): error: in "RPTRST_AND_RPTSOL_SOLUTION": Must write restart information for included step 11
${ROOT}/opm-common/tests/parser/RestartConfigTests.cpp(1135): error: in "RPTRST_AND_RPTSOL_SOLUTION": Must write restart information for included step 14
${ROOT}/opm-common/tests/parser/RestartConfigTests.cpp(1135): error: in "RPTRST_AND_RPTSOL_SOLUTION": Must write restart information for included step 17
```
If I move the `RPTRST` keyword to the `SCHEDULE` section the test mostly passes, except for report step 1.  It's not fully clear to me whether there should be restart output for report step 1 in this case.  It is arguably the first real report step in the first simulated month whence
```
RPTRST
  'BASIC=5' 'FREQ=6
/
```
should generate output.

---

```diff
diff --git a/tests/parser/RestartConfigTests.cpp b/tests/parser/RestartConfigTests.cpp
index 4171d98ef..cdebcdb03 100644
--- a/tests/parser/RestartConfigTests.cpp
+++ b/tests/parser/RestartConfigTests.cpp
@@ -1083,3 +1083,56 @@ BOOST_AUTO_TEST_CASE(RESTART_SAVE) {
 
 }
 
+BOOST_AUTO_TEST_CASE(RPTRST_AND_RPTSOL_SOLUTION)
+{
+    const auto input = std::string { R"(RUNSPEC
+DIMENS
+  10 10 10 /
+START
+  6 JLY 2020 /
+SOLUTION
+RPTSOL
+  'RESTART=2' 'FIP=3' 'FIPRESV' 'THPRES' /
+RPTRST
+  'BASIC=5' 'FREQ=6' 'CONV=10' /
+SCHEDULE
+DATES
+  7 'JLY' 2020 / -- WRITE ( 1)
+ 10 'JLY' 2020 /          ( 2)
+ 20 'JLY' 2020 /          ( 3)
+ 30 'JLY' 2020 /          ( 4)
+  5 'AUG' 2020 /          ( 5)
+ 20 'AUG' 2020 /          ( 6)
+  5 'SEP' 2020 /          ( 7)
+  1 'OCT' 2020 /          ( 8)
+  1 'NOV' 2020 /          ( 9)
+  1 'DEC' 2020 /          (10)
+  5 'JAN' 2021 / -- WRITE (11)
+  1 'FEB' 2021 /          (12)
+ 17 'MAY' 2021 /          (13)
+  6 'JLY' 2021 / -- WRITE (14)
+  1 'DEC' 2021 /          (15)
+ 31 'DEC' 2021 /          (16)
+ 21 'JAN' 2022 / -- WRITE (17)
+ 31 'JAN' 2022 /          (18)
+/
+END
+)" };
+
+    const auto deck   = Parser{}.parseString(input);
+    const auto tm     = TimeMap { deck };
+    const auto rstcfg = RestartConfig { tm, deck };
+
+    BOOST_REQUIRE_EQUAL(tm.size(), std::size_t{19});
+    BOOST_REQUIRE_EQUAL(tm.last(), std::size_t{18});
+
+    for (const std::size_t stepID : { 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 13, 15, 16, 18 }) {
+        BOOST_CHECK_MESSAGE(! rstcfg.getWriteRestartFile(stepID),
+                            "Must not write restart information for excluded step " << stepID);
+    }
+
+    for (const std::size_t stepID : { 0, 1, 11, 14, 17 }) {
+        BOOST_CHECK_MESSAGE(rstcfg.getWriteRestartFile(stepID),
+                            "Must write restart information for included step " << stepID);
+    }
+}
```

**Comment by bska** [comment]
@joakim-hove: Would you be able to have a look at this or do you need additional information?

**Comment by joakim-hove** [comment]
Holiday has interweened - I'm looking into this now.

**Comment by bska** [comment]
Solved in #1876.
